server/services/model_service.py

```diff
@@ -0,0 +1,127 @@
+"""Model configuration service for multi-model support.
+
+This service handles model resolution and provides utilities
+for working with multi-model configurations.
+"""
+
+import sys
+from pathlib import Path
+from typing import Any
+
+# Add repo root to path for config imports
+repo_root = Path(__file__).parent.parent.parent
+sys.path.insert(0, str(repo_root))
+
+from config.datamodel import LlamaFarmConfig, Provider, PromptFormat, Lemonade  # noqa: E402
+from core.logging import FastAPIStructLogger  # noqa: E402
+
+logger = FastAPIStructLogger(__name__)
+
+
+class ModelConfig:
```

Comment:
This whole class is no good. There should be a Model  config coming from the shared config.datamodel generated by the schema